bolt-design-system/bolt
# Grid enhancements

## Jira

http://vjira2:8080/browse/BDS-829

## Summary

New grid component that handles complex layouts with versatile breakpoint options.

## Details

1. Built on top of the existing new grid that was not officially released.
2. Narrowed props down to column-start, column-span, row-start, row-span, and valign for each grid items.
3. The main grid container has gutter and row-gutter props to handle vertical and horizontal spacing.
4. Breakpoint specific options can allow the items to transform freely.

## How to test

1. Run the branch locally.
2. Check all the grid docs and make sure it's easy to understand.
3. Try to follow the docs and create some dummy layouts, see if the grid is working as you'd expect.

**Note: You can use this image as an reference to create some common layouts for testing.**

![grid](https://user-images.githubusercontent.com/3027663/51706729-5842e500-1fed-11e9-8f47-8322749584e5.png)
